sample/composeApp/src/commonMain/kotlin/com/mifos/passcode/sample/passcode/PasscodeRepository.kt

```diff
@@ -2,24 +2,37 @@ package com.mifos.passcode.sample.passcode
 
 import com.mifos.passcode.sample.kmpDataStore.PreferenceDataStore
 
-private const val PASSCODE_INFO_KEY = "passcodeInfo"
-
+private const val PASSCODE_INFO_KEY = "mifos.passcode.passcode"
+private const val PASSCODE_LENGTH = "mifos.passcode.length"
 class PasscodeRepository (
     private val source: PreferenceDataStore,
 ){
     fun getPasscode(): String {
         return source.getSavedData(PASSCODE_INFO_KEY,"")
     }
 
+    fun getPasscodeLength(): Int {
+        val passcodeLength = source.getSavedData(PASSCODE_LENGTH, "")
+        return if(passcodeLength.isNotEmpty()) passcodeLength.toInt() else 0
+    }
+
     fun savePasscode(passcode: String) {
         source.putData(
             PASSCODE_INFO_KEY,
             passcode
         )
     }
 
+    fun savePasscodeLength(passcodeLength: Int){
+        source.putData(
+            PASSCODE_LENGTH,
+            passcodeLength.toString()
+        )
```

Comment:
why we are changing it to `string` can't we store `Int`. so in that we you would not need to change it from `string` back to `Int` like you are doing above for getting it. It will help avoiding making extra conversions